### PR TITLE
bug/core-js-fixes

### PR DESCRIPTION
**Pull request by cliffmeyers**
# Description
* Minor fix on core-js to prevent `Utils.clone` from bombing out when passing null / undefined
* Enhance `Fetch` so the response body from an error response is parsed and attached to underlying error.  cc @imeredith as this one is worth soliciting input...
  * The change didn't seem to cause any issues but wanted to get your eyes on it.
  * Wondering if it's worth combining `checkResult` and `parseJSON` into a single method since the logic in new `parseErrorJSON` is a bit duplicative and also lacks some error handling (see below).
  * A ways back I had to add some error handling into the original `parseJSON` func to deal with an apparent Chrome bug on empty 200 responses. I want to test and see if this is still the case to see if that code can be removed; otherwise I may need to add it to the new function as well or combine as i mentioned above.

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by imeredith** [comment]
LGTM :bee:

File: blueocean-core-js/src/js/utils.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ const cleanSlashes = (url: string) => {
 export default {
     cleanSlashes,
     clone(obj: Object) {
+        if (!obj) return obj;
```
Comment (michaelneale) [inline blueocean-core-js/src/js/utils.js]: wouldn't this just be returning null?

Comment (imeredith) [inline blueocean-core-js/src/js/utils.js]: or undefined

Comment (michaelneale) [inline blueocean-core-js/src/js/utils.js]: gotcha

Comment (cliffmeyers) [inline blueocean-core-js/src/js/utils.js]: The intent of the method is to clone, so I am not sure what else would be appropriate to return. Empty object doesn't quite seem right, can't think of anything else TBH.

**Comment by michaelneale** [comment]
@cliffmeyers safe to go in? Checked with ATH? if so - go for it, small change :bee:

**Comment by michaelneale** [comment]
(Please ensure it doesn't break the ATH, as want to cut a beta)

**Comment by tfennelly** [comment]
LGTM 🐝

File: blueocean-core-js/src/js/fetch.js [hunk 2]
```diff
@@ -127,7 +145,7 @@ export const FetchFunctions = {
             const future = isoFetch(url, FetchFunctions.sameOriginFetchOption(fetchOptions))
                 .then(FetchFunctions.checkRefreshHeader)
                 .then(FetchFunctions.checkStatus)
-                .then(FetchFunctions.parseJSON);
+                .then(FetchFunctions.parseJSON, FetchFunctions.parseErrorJson);
```
Comment (scherler) [inline blueocean-core-js/src/js/fetch.js]: Why not a new line?

`
.then(FetchFunctions.parseJSON)
.then(FetchFunctions.parseErrorJson)
 `

**Comment by michaelneale** [comment]
@cliffmeyers is this one still relevant?

**Comment by cliffmeyers** [comment]
ATH passed locally.

```
2017-01-04 11:44:06,717 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - OK. 934  total assertions passed. (9m 25s)
2017-01-04 11:44:06,789 [Thread-3] INFO   (InputStreamHandler.java:47) com.github.eirslett.maven.plugins.frontend.lib.DefaultGulpRunner - [11:44:06] Finished 'default' after 9.42 min
**** NightwatchRunner - No failures.
```

**Pull request merged**